torch/lib/c10d/Utils.cpp

```diff
@@ -231,6 +241,16 @@ int connect(
         if (!wait || (!anyRefused && !anyReset)) {
           throw;
         }
+
+        // if a timeout is specified, check time elapsed to see if we need to
+        // timeout. A timeout is specified if timeout != kNoTimeout.
+        if (timeout != kNoTimeout) {
+          const auto elapsed =
+              std::chrono::high_resolution_clock::now() - start;
+          if (elapsed > timeout) {
+            throw std::runtime_error("connect() timed out");
```

Comment:
The error msg now reads the same as the one we throw when `poll` times out. Should we make this a const string somewhere?

Reply:
Yes, good idea. We like to use `kSomeName` for globals, but this one can be kept local only in an anonymous namespace (equivalent of static global), since it's only used in this file.